inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp

```diff
@@ -327,6 +327,8 @@ void MKLDNNGraph::InitGraph() {
 
     InitEdges();
 
+    InitInPlaceForNodes();
```

Comment:
What do you think, may be it would be better to include this subroutine into the `InitOptimalPrimitiveDescriptors` stage and inline the code? On the other hand, it would be even better to move this caching responsibility onto Node class itself, at leas from the code reliability point of view.

Reply:
After `InitOptimalPrimitiveDescriptors` we have `InitEdges` call where may be created new nodes: `reorder` or `convert`. So I added `InitInPlaceForNodes` call after this function. If I didn't smth missed, these nodes always have `inplace = false`. But I think it will be better when we would call `InitInPlaceForNodes` after all nodes creations. So may be we should call `InitInPlaceForNodes` after `ApplyImplSpecificGraphOptimizations`.


What about moving of caching responsibility onto Node class, I agree that it would be better. But it's not clear when exactly it is necessary to initialize `inplace`. If we will initialize this field in `MKLDNNNode::InitOptimalPrimitiveDescriptor` or `MKLDNNNode::CreatePrimitive` (after all nodes creations), we should to rewrite all these methods. It's not good. So we have question, when we should init `inplace` in Node class. Or did I not fully understand you about this?

Reply:
Ok, I agree about the first part. Now about caching. One such solution might be, for example, storing  the `inplace` flag not as a bool value, but as an enum with three values (yes, no, unknow(default)), when it is unknown (for the first `isInplace()` call) we actually do the check and define the flag. Another important thing is that the flag must be kept in the "private" section so that it cannot be changed in derived classes.